src/pages/additem/Additem.js

```diff
@@ -0,0 +1,148 @@
+import styles from "./Additem.module.css";
+import { useForm } from "./hooks/useForm";
+import ImageUpload from "./components/ImageUpload";
+import TextInputField from "./components/TextInputField";
+import TagInput from "./components/TagInput";
+import { useValidation } from "./hooks/useValidation";
+import { useRef } from "react";
+import { formatWithCommas } from "./utils/formatUtils";
+
+export default function Additem() {
+  const productNameRef = useRef();
+  const productDescriptionRef = useRef();
+  const productPriceRef = useRef();
+  const { validations } = useValidation();
+  const { data, errors, handleChange, handleBlur, handleSubmit } = useForm({
+    initialValues: {
+      productName: "",
+      productDescription: "",
+      productPrice: "",
+      productTag: [],
+      uploadImage: null,
+    },
+    validations,
+    onSubmit: (formData) => {
+      // 콤마 제거 + 숫자형으로 변환
+      const numericPrice = Number(formData.productPrice.replace(/,/g, ""));
+      const payload = {
+        ...formData,
+        productPrice: numericPrice,
+      };
+      console.log("제출 데이터:", payload);
+    },
+  });
+
+  const handlePriceChange = (e) => {
+    const formatted = formatWithCommas(e.target.value);
+    // 1) React state 갱신
+    handleChange(
+      "productPrice",
+      formatWithCommas
+    )({ target: { value: formatted } });
+    // 2) DOM value도 즉시 갱신
+    productPriceRef.current.value = formatted;
+  };
+
+  // 필수 필드만 모두 채워졌는지 확인
+  const isSubmitEnabled =
+    data.productName.trim() !== "" &&
+    data.productDescription.trim() !== "" &&
+    data.productPrice.trim() !== "" &&
+    Array.isArray(data.productTag) &&
+    data.productTag.length > 0;
```

Comment:
위 코멘트처럼 getter함수를 useForm 훅에서 내려받는다면 이런 형태가 되겠죠?

```jsx
  const {
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    getFieldValue,
    setFieldValue,
  } = useForm({
    initialValues: {
      productName: "",
      productDescription: "",
      productPrice: "",
      productTag: [],
      uploadImage: null,
    },
    validations,
    onSubmit: (formData) => {
      // 콤마 제거 + 숫자형으로 변환
      const numericPrice = Number(formData.productPrice.replace(/,/g, ""));
      const payload = {
        ...formData,
        productPrice: numericPrice,
      };
      console.log("제출 데이터:", payload);
    },
  });
```
예시)
```suggestion
  const isSubmitEnabled =
    getFieldValue("productName")?.trim() !== "" &&
    getFieldValue("productDescription")?.trim() !== "" &&
    getFieldValue("productPrice")?.trim() !== "" &&
    Array.isArray(getFieldValue("productTag")) &&
    getFieldValue("productTag").length > 0;
```